hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -1164,4 +1197,62 @@ private String getStateMessage() {
                 + " loadedOnCreate=" + loadedOnCreate + " loadedOnPreMigration=" + loadedOnPreMigration
                 + " isLoaded=" + isLoaded();
     }
+
+    @Override
+    public void postProcess(Data dataKey, Object oldValue, Object value, Object mergingValue,
+                            EntryEventType eventType, boolean replicateOverWAN,
+                            String callerUuid, Address callerAddress) {
+
+        if (eventType == null) {
+            eventType = oldValue == null ? ADDED : UPDATED;
+        }
+
+        Object endValue = extractEndValue(dataKey, value);
+        mapServiceContext.interceptAfterPut(name, endValue);
```

Comment:
Previously this method was called with the `value` and not the end value. Is this an acceptable change?